### PR TITLE
Mib 115

### PR DESCRIPTION
**Pull request by PasGat**

### CODE REVIEW

File: app/uk/gov/hmrc/merchandiseinbaggagefrontend/controllers/CustomsAgentController.scala [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import uk.gov.hmrc.merchandiseinbaggagefrontend.model.core.YesNo
+import uk.gov.hmrc.merchandiseinbaggagefrontend.model.core.YesNo.{No, Yes}
+import uk.gov.hmrc.merchandiseinbaggagefrontend.views.html.CustomsAgentView
+
+import scala.concurrent.ExecutionContext
+
+class CustomsAgentController @Inject()(
+                                        override val controllerComponents: MessagesControllerComponents,
+                                        actionProvider: DeclarationJourneyActionProvider,
+                                        formProvider: CustomAgentFormProvider,
```
Comment (jordanrowe) [inline app/uk/gov/hmrc/merchandiseinbaggagefrontend/controllers/CustomsAgentController.scala:33]: We moved away from this pattern now.

Forms are declared as objects with val form which can be referenced directly from import

File: conf/messages [hunk 2]
```diff
@@ -205,6 +204,13 @@ cannotUseService.a = Make a full custom declaration
 cannotUseService.href = https://www.gov.uk/guidance/making-a-full-import-declaration
 
 
+# Custom Agent
+customsDeclares.title = Are you a customs agent?
```
Comment (jordanrowe) [inline conf/messages:208]: should prefix not be customsAgent as that is the name of the View/Controller etc

File: app/uk/gov/hmrc/merchandiseinbaggagefrontend/model/core/Declaration.scala [hunk 3]
```diff
@@ -201,7 +201,7 @@ case class DeclarationJourney(sessionId: SessionId,
         maybeExciseOrRestrictedGoods.contains(false) &&
         maybeValueWeightOfGoodsExceedsThreshold.contains(false) &&
         journeyDetailsCompleteAndDeclarationRequired &&
-        (maybeCustomsAgent.isDefined || maybeIsACustomsAgent.contains(false))
+        maybeCustomsAgent.isDefined
```
Comment (PaulHodgson) [inline app/uk/gov/hmrc/merchandiseinbaggagefrontend/model/core/Declaration.scala:204]: Not sure about this

To complete a declaration you'd only need to have a defined customs agent of you answer 'yes' to the question `maybeIsACustomsAgent`.

Declaration can be complete if you answer 'No' to maybeIsACustomsAgent; if you answer 'Yes' then maybeCustomsAgent.isDefined must be true.

So it looks like there's a missing test in DeclarationSpec (my bad).

Comment (PaulHodgson) [reply app/uk/gov/hmrc/merchandiseinbaggagefrontend/model/core/Declaration.scala:204]: Doh! Missed the guard on line 182

**Review by PaulHodgson — APPROVED** [review]

**Pull request merged**